boot/core/src/boot/task_helpers/notify.clj

```diff
@@ -22,29 +23,72 @@
     (io/copy (io/input-stream (io/resource "boot-logo-3.png")) f)
     (.getAbsolutePath f)))
 
+(defn sh-with-timeout [& args]
+  (try
+    (apply util/dosh-timed 1000 args)
+    0
+    (catch Exception e
+      (util/print-ex e)
+      1)))
+
 (defn- ^{:boot/from :jeluard/boot-notify} program-exists?
   [s]
-  (= 0 (:exit (shell/sh "sh" "-c" (format "command -v %s" s)))))
+  (= 0 (sh-with-timeout "sh" "-c" (format "command -v %s >/dev/null" s))))
+
+(defn- try-notify-with-shell-program [program-name & args]
+  (and
+   (program-exists? program-name)
```

Comment:
I mean, yeah, just handle the case gracefully so that we know when there is a program missing